crawler/containers_logs_linker.py

```diff
@@ -0,0 +1,120 @@
+from containers import get_containers
+from base_crawler import BaseCrawler
+from worker import Worker
+import argparse
+import os
+import misc
+
+
+class DockerContainersLogsLinker(BaseCrawler):
+    """
+    Class used to maintain symlinks to container log files. The idea with this
+    is to symlink all log files of interest (from all containers of interest)
+    to some known directory in the host. Then point some log collector like
+    logstash to it (and get container logs).
+    """
+
+    def __init__(self,
+                 environment='cloudsight',
+                 user_list='ALL',
+                 host_namespace=''):
+
+        self.containers_list = set()
+        self.new = set()
+        self.deleted = set()
+        self.environment = environment
+        self.host_namespace = host_namespace
+        self.user_list = user_list
+
+    def update_containers_list(self):
+        """
+        Actually poll for new containers. This updates the list of new and
+        deleted containers, in self.new and self.deleted.
+        :return: None
+        """
+        curr_containers = set(
+            get_containers(
+                environment=self.environment,
+                user_list=self.user_list,
+                host_namespace=self.host_namespace))
+        self.new = curr_containers - self.containers_list
+        self.deleted = self.containers_list - curr_containers
+        self.containers_list = curr_containers
+
+    def link_containers(self):
+        for container in self.deleted:
+            container.unlink_logfiles()
+        for container in self.new:
+            container.link_logfiles()
+
+    def crawl(self):
+        self.update_containers_list()
+        self.link_containers()
+        return []
+
+
+if __name__ == '__main__':
```

Comment:
Maintaining logs wasn't fitting well with the main we already have. It's was like an extra weird feature, so decided to move it to its own main. The test for it (and how to use it) is `tests/functional/test_functional_logs_linker.py`.